src/components/Map.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react';
+import GoogleMapReact from 'google-map-react';
+
+import withResource from './withResource';
+import { TResource } from '../types';
+
+type Props = { data: TResource };
+
+const boulderCoordinates = {
+  lat: 40.0156852,
+  lng: -105.2792069
+};
+
+class Map extends Component<any, any> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      directionsAreToggled: false
+    };
+  }
+
+  calculateAndDisplayRoute = (
+    directionsService: any,
+    directionsDisplay: any,
+    resourceAsDestination: any
+  ) => {
+    const userLocation = this.props.currentUser.location;
+
+    // remove default markers A and B (origin/destination)
+    directionsDisplay.setOptions({ suppressMarkers: true });
+
+    directionsService.route(
+      {
+        origin: userLocation,
+        destination: {
+          lat: resourceAsDestination.lat,
+          lng: resourceAsDestination.lng
+        },
+        travelMode: 'TRANSIT'
+      },
+      function(response: any, status: any) {
+        if (status === 'OK') {
+          directionsDisplay.setDirections(response);
+        } else {
+          console.log('Directions request failed due to ' + status);
+        }
+      }
+    );
+  };
+
+  toggleDirections = (resource: any) => {
+    const {
+      map,
+      directionsDisplay,
+      directionsService,
+      directionsAreToggled
+    } = this.state;
+
+    if (!directionsAreToggled) {
+      directionsDisplay.setMap(map);
+
+      this.calculateAndDisplayRoute(
+        directionsService,
+        directionsDisplay,
+        resource
+      );
+    } else {
+      directionsDisplay.setMap(null);
+    }
+
+    this.setState({ directionsAreToggled: !directionsAreToggled });
+  };
+
+  createMarkers = (map: any, maps: any) => {
+    const resources = this.props.resources;
+    resources.forEach((resource: any) => {
+      let marker = new maps.Marker({
+        map: map,
+        title: resource.name,
+        position: {
+          lat: resource.lat,
+          lng: resource.lng
+        }
+      });
+
+      let infoWindow = new maps.InfoWindow({
+        content:
+          '<div><b>' +
+          resource.name +
+          '</b><br>' +
+          resource.category +
+          '<br>' +
+          resource.address +
+          '<br><br>' +
+          '<b>For</b>: ' +
+          resource.audience +
+          '<br>' +
+          '<b>Offers</b>: ' +
+          resource.services.join(', ') +
+          '</div>'
+        // will need to include if its open/hours (unsure of how resource.hours will be formatted)
```

Comment:
```suggestion
        // TODO: Include if its open/hours (unsure of how resource.hours will be formatted)
        // TODO: Use a custom component instead of an infoWindow
```